Tests/Marketplace/search_and_install_packs.py

```diff
@@ -690,18 +690,27 @@ def search_and_install_packs_and_their_dependencies(pack_ids: list,
 
     lock = Lock()
 
-    with ThreadPoolExecutor(max_workers=130) as pool:
+    if not install_packs_one_by_one:
+        with ThreadPoolExecutor(max_workers=130) as pool:
+            for pack_id in pack_ids:
+                if is_pack_hidden(pack_id):
+                    logging.debug(f'pack {pack_id} is hidden, skipping installation and not searching for dependencies')
+                    continue
+                pool.submit(search_pack_and_its_dependencies,
```

Comment:
You're not waiting for them to finish?

Reply:
`with ThreadPoolExecutor` context manager handles it. (waiting to finish on exit).